dds_web/scheduled_tasks.py

```diff
@@ -254,14 +288,41 @@ def monthly_usage():
                     )
                     .with_for_update()
                 ):
-                    proj_bhours, proj_cost = UserProjects.project_usage(project)
+                    usage_data[unit.safespring_name][project.id] = {}
+                    proj_bhours, proj_cost = UserProjects.project_usage(project=project)
                     scheduler.app.logger.info(
                         "Current total usage for project %s is %s bhours, and total cost is %s kr",
-                        project.public_id,
+                        project.id,
                         proj_bhours,
                         proj_cost,
                     )
+                    usage_data[unit.safespring_name][project.id]["usage"] = proj_bhours
+                    usage_data[unit.safespring_name][project.id]["cost"] = proj_cost
+                    unit_usage += proj_bhours
+                    unit_cost += proj_cost
+                usage_data[unit.safespring_name]["Unit usage"] = unit_usage
+                usage_data[unit.safespring_name]["Unit cost"] = unit_cost
         except (sqlalchemy.exc.OperationalError, sqlalchemy.exc.SQLAlchemyError) as err:
             flask.current_app.logger.exception(err)
             db.session.rollback()
             raise
+
+        # use the usage dictionary to make a DB record
+        for unit, data in usage_data.items():
+            if data["Unit usage"] != 0:
+                scheduler.app.logger.info(f'Total usage for unit {unit} is: {data["Unit usage"]}')
+                for project, pdata in data.items():
+                    if isinstance(project, int):
+                        new_record = models.Usage(
+                            project_id=project,
+                            usage=pdata["usage"],
+                            cost=pdata["cost"],
+                            time_collected=current_time(),
+                        )
```

Comment:
I'm thinking we could move this to the previous for loop and just create the usage row after the calculation of the usage? Instead of doing another iteration through the items and saving to a dict

Reply:
Yeah, I agree

Reply:
Also, I think you can remove the `usage_data` parts now since you're saving the info to the db straight away anyway and increasing the total usage, so `usage_data` is not needed.

Reply:
Yes, I was thinking about this already yesterday; it is used only for the percentages calculation, and this is something we can do by querying the DB in the billing job.